app/script/view_model/ConversationInputEmojiViewModel.js

```diff
@@ -0,0 +1,177 @@
+/*
+ * Wire
+ * Copyright (C) 2017 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+z.ViewModel = z.ViewModel || {};
+
+z.ViewModel.ConversationInputEmojiViewModel = class ConversationInputEmojiViewModel {
+  constructor() {
+    this.emoji_list = $('<div class="conversation-input-emoji-list" />');
+    this.emoji_dict = undefined;
+    this.emoji_start_pos = -1;
+
+    fetch('/image/emoji.tsv')
+      .then((response) => response.text())
+      .then((text) => {
+        this.emoji_dict = text.split('\n');
+      });
+  }
+
+  static get EMOJI_LIST_LENGTH() {
+    return 5;
+  }
+
+  static get MIN_QUERY_LENGTH() {
+    return 2;
```

Comment:
My dream is to include this rule pretty soon: http://eslint.org/docs/rules/no-magic-numbers and while it feels like overkill I'm sure it will throw a warning.. any other ideas?

(obviously excluding `0s` and `1s` I guess)